### DISCUSSION TITLE
How can we stop duplicate queue requests

### DISCUSSION THREAD

**Discussion by chandamotiyani** [craftcms/cms#9110] [Ideas]
Hi,

We do send a request to an API through the queue in craft site, but for some unknown reasons it is being sent twice or thrice at the same time with no fraction of time difference, and its causing duplications of records on our client's system.

Is there any way of checking that if the same request has been sent previously and shouldn't be sent again?

In the attached screenshot you could see the log - **create yalumba customer**  has been sent twice at the same time

<img width="1042" alt="Screen Shot 2020-09-28 at 10 31 30 am" src="https://user-images.githubusercontent.com/16190427/94380870-6bb93280-0176-11eb-9a10-a480515b8445.png">

Thank you

**Comment by chandamotiyani** [comment]
@brandonkelly is there anything to stop duplicate queue requests from happening?

**Comment by brandonkelly** [comment]
I’m guessing it’s due to a race condition due to the initial job not getting run quickly enough. If you set up a daemonized queue worker you should be able to reduce the likelihood of it happening, but unfortunately there’s no way to prevent duplicate jobs across all queue drivers.

**Comment by Mosnar** [comment]
@chandamotiyani If you're not able to get to the bottom of it, you _could_ calculate a hash for the JSON payload and store a short-lived flag in the cache. Something like this:

```php
$jsonPayload = "{'foo': 'bar'}";
$hashedJsonPayload = md5($jsonPayload);
$cacheKey = 'yalumba:' . $hashedJsonPayload;
if (!Craft::$app->cache->get($cacheKey)) {
    // Expire in 1 day
    Craft::$app->cache->set($cacheKey, true, 86400);
    // TODO: Send data to API
}
```

**However** - You're better off fixing the source of the problem, like Brandon suggested.
